### ISSUE TITLE
demo script throws an exception

### ISSUE THREAD

**Issue by vlas-sokolov** [JohannesBuchner/PyMultiNest#76]
Hi @JohannesBuchner! I'm trying to run one of the example files, `pymultinest_demo.py`, and it seems to run fairly smoothly, except that one of the threads throws an error in the middle of the run. Is this normal? Also, the plotting routines brake down later on, probably because the output in `chains/` isn't complete?

Here's a full output that the script generates:
http://pastebin.com/GV2JEdYQ

**Comment by JohannesBuchner** [comment]
Could you try replacing the first part of the script with 

```
from sys import platform
if not os.path.exists("chains"): os.mkdir("chains")
def show(filepath): 
	""" open the output (pdf) file for the user """
	if os.name == 'mac' or platform == 'darwin': subprocess.call(('open', filepath))
	elif os.name == 'nt' or platform == 'win32': os.startfile(filepath)
	elif platform.startswith('linux') : subprocess.call(('xdg-open', filepath))
```

**Comment by JohannesBuchner** [comment]
Also try the minimal demo, it has fewer bells and whistles.

**Comment by vlas-sokolov** [comment]
Ah I see. The on-screen output made it look like the error came from within the call to `pymultinest.run()` but it wasn't. Updating the show function solved that first error.

Still getting an error message though every time the script tries to plot conditional probability plots. Below is a slightly modified minimal example that reproduces the error for me:

```python
import pymultinest
import math, os
if not os.path.exists("chains"): os.mkdir("chains")

def myprior(cube, ndim, nparams):
    for i in range(ndim):
        cube[i] = cube[i] * 10 * math.pi

def myloglike(cube, ndim, nparams):
    chi = 1.
    for i in range(ndim):
        chi *= math.cos(cube[i] / 2.)
    return math.pow(2. + chi, 5)

parameters = ["x", "y"]
n_params = len(parameters)

pymultinest.run(myloglike, myprior, n_params, resume = True, verbose = True)

a = pymultinest.Analyzer(n_params = n_params, outputfiles_basename='chains/2-')
p = pymultinest.PlotMarginalModes(a)
p.plot_conditional(0, 1)
```

```python
ValueError: Contour levels must be increasing
```

**Comment by paulkerry1** [comment]
Hi

The "ValueError: Contour levels must be increasing" looks like it is due to changes with contour within matplotlib (i.e. https://github.com/matplotlib/matplotlib/pull/5485 ).

This is with matplotlib 1.5.3 under 64-bit linux.

I was wondering if a fix was available?

Cheers and keep up the good work!
Paul

**Comment by JohannesBuchner** [comment]
Hi @paulkerry1, thanks for the report.

Can you try changing in pymultinest/plot.py line 142 "levels" to "levels[::-1]"
Do you know if increasing levels also work in earlier matplotlib versions?

Johannes

**Comment by paulkerry1** [comment]
Changing "levels" to "levels[::-1]" works perfectly for the pymultinest_demo.py and for the other python scripts our people are using.

Sorry I don't have any earlier versions of matplotlib to test...

Cheers
Paul.

**Comment by JohannesBuchner** [comment]
Fixed in commit #f18c8e4
